woocommerce/woocommerce-ios
# Product selector: add single selection mode and simple selection handling mode, attempt 2.

Closes: https://github.com/woocommerce/woocommerce-ios/issues/11585


## Description

This is a second attempt after a previous PR #11589 ended up creating a bit of too complex solution.

The difference between this PR and the old one are:
- We're now using the existing `multipleSelectionsEnabled` parameter of `ProductRow` to support single selection.
- This does not touch the already complex `ProductSelectorViewModel` (props @itsmeichigo )
- This now follows the latest Blaze i3 design where we remove the "Done" button and checkbox next to each product. This also helps simplify the code quite a bit.
- This PR now supports navigating into Product Selector using Navigation Controller, to better match Blaze i3 design requirement.
- This PR now connects the Dashboard Blaze Create Campaign button to open Product Selector.

Known issue:
- This does not connect Product Selector with the Blaze list screen's campaign creation. It will be handled in a separate PR after we finalize the correct navigation architecture in this PR.

## Testing instructions

1. Start with a site with multiple products,
2. in My Store, tap "Create Campaign" button,
3. ensure product selector is shown,
4. tap a Product, ensure the new campaign creation screen is shown.
5. Go back, ensure selector is shown again.
6. Tap variation or bundle product, ensure the new campaign creation screen is shown as well,
7. Go back, ensure selector is shown again,
8. Press Cancel, ensure the selector is now hidden.

## Video
This shows testing simple product, then bundle, then variable product, and they all open campaign creation properly. 

https://github.com/woocommerce/woocommerce-ios/assets/266376/8cb1ada1-c024-4160-99b0-e34a09dd9bd2
